### DISCUSSION TITLE
Resolving a URL to a dynamic route pattern

### DISCUSSION THREAD

**Discussion by ericbiewener** [vercel/next.js#48108] [Help]
### Summary

We're using Next.js 12.2.5 and have a custom server wrapping it on the backend. We need our custom server to be able to identify which Next.js dynamic route a given URL maps to. For example, something like:

```tsx
getDynamicRouteForURL("https://example.org/items/10/edit") // returns `/items/[id]/edit`
```

I don't see any documented API, nor can I identify a clear function in the codebase that does this. I know Next outputs a `routes-manifest.json` artifact to the build directory. Perhaps testing URLs given the configuration found in that file is sufficient? I was hoping to confirm that by seeing that the same thing is done in the Next.js codebase, but the code that appears to do this is in [the `Router.execute` method](https://github.com/vercel/next.js/blob/v12.2.5/packages/next/server/router.ts#L131), which is also doing a lot of other stuff and therefore doesn't offer the straightforward confirmation I was hoping for.

Looking for anything that can help me confidently reverse-engineer the Next's URL-to-route matching algorithm.

**Comment by osdiab** [comment]
Did you ever figure something out regarding this? I'm having the same issue, this time for trying to figure out how to correctly provide i18n namespaces to my frontend by route. Something like this would be very helpful.

**Reply by ericbiewener** [reply]
No, sorry. Instead I managed to figure out a way to implement the functionality I needed _inside_ the react SSR lifecycle. That gives me access to the Next.js router instance, which has the dynamic route path available.

**Reply by krewetka** [reply]
Could you provide a bit more details how you did it @ericbiewener ?
